templates/repo/diff/box.tmpl

```diff
@@ -107,14 +108,15 @@
 										{{end}}
 									</div>
 									<span class="file gt-mono"><a class="muted file-link" title="{{if $file.IsRenamed}}{{$file.OldName}} &rarr; {{end}}{{$file.Name}}" href="#diff-{{$file.NameHash}}">{{if $file.IsRenamed}}{{$file.OldName}} &rarr; {{end}}{{$file.Name}}</a>{{if .IsLFSFile}} ({{$.locale.Tr "repo.stored_lfs"}}){{end}}</span>
+									<button class="ui button button-link gt-p-3" data-clipboard-text="{{$file.Name}}">{{svg "octicon-copy"}}</button>
```

Comment:
We could even think of hiding the button on mobile devices as you don't really need it there, and space is scarce already there.

Reply:
Mobile layout does need some tweaks 😆 

<img width="437" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/57373938-ca50-469b-8c6b-bc8d7aa1eab5">